test/e2e/app/app.go

```diff
@@ -719,18 +718,14 @@ func (app *Application) checkHeightAndExtensions(isPrepareProcessProposal bool,
 
 func (app *Application) storeValidator(valUpdate *abci.ValidatorUpdate) error {
 	// Store validator data to verify extensions
-	pubKey, err := cryptoenc.PubKeyFromProto(valUpdate.PubKey)
+	pubKey, err := cryptoenc.PubKeyFromTypeAndBytes(valUpdate.PubKeyType, valUpdate.PubKeyBytes)
 	if err != nil {
 		return err
 	}
 	addr := pubKey.Address().String()
 	if valUpdate.Power > 0 {
-		pubKeyBytes, err := valUpdate.PubKey.Marshal()
```

Comment:
I still think you need to marshall here, otherwise you're losing key type information.
Alternatively you can also include the key type concatenated in in line 728 below. But marshalling is a lot simpler